src/analysis/complex/real_deriv.lean

```diff
@@ -62,3 +62,123 @@ times_cont_diff_iff_times_cont_diff_at.2 $ λ x,
+    conformal at a point `z` if it is holomorphic at that point -/
+lemma conformal_at_of_holomorph (hf' : fderiv ℝ f z ≠ 0) (h : differentiable_at ℂ f z) :
+  conformal_at f z :=
+begin
+  have : differentiable_at ℝ f z := h.restrict_scalars ℝ,
+  rw [conformal_at_iff_is_conformal_map_fderiv],
+  rw [differentiable_at_iff_exists_linear_map ℝ this] at h;
+    [skip, apply_instance, apply_instance, apply_instance],
+  rcases h with ⟨map, hmap⟩,
+  rw ← hmap,
+  refine is_conformal_map_complex_linear _,
+  contrapose! hf' with w,
+  ext1,
+  simp only [← hmap, coe_restrict_scalars', w, continuous_linear_map.zero_apply],
+end
+
+/-- A real differentiable function of the complex plane into some complex normed space `E` is
+    conformal at a point `z` if it is antiholomorphic at that point -/
+lemma conformal_at_of_antiholomorph (hf' : fderiv ℝ f z ≠ 0)
+  (h : differentiable_at ℂ (f ∘ conj) (conj z)) : conformal_at f z :=
+begin
+  have minor₁ : f ∘ conj ∘ conj = f := by ext1; simp only [function.comp_app, conj_conj],
+  have minor₂ : differentiable_at ℝ f z :=
+    minor₁ ▸ (h.restrict_scalars ℝ).comp z (has_fderiv_at_conj z).differentiable_at,
+  rw [conformal_at_iff_is_conformal_map_fderiv],
+  rw [antiholomorph_at_iff_exists_complex_linear_conj minor₂] at h,
+  rcases h with ⟨map, hmap⟩,
+  have minor₃ : fderiv ℝ f z = (map.restrict_scalars ℝ).comp ↑conj_cle,
+  { ext1,
+    rw hmap,
+    simp only [coe_comp', function.comp_app, conj_cle.coe_coe, conj_cle_apply, conj_conj], },
+  rw minor₃,
+  refine is_conformal_map_complex_linear_conj _,
+  contrapose! hf' with w,
+  rw minor₃,
+  ext1,
+  simp only [coe_comp', function.comp_app, conj_cle.coe_coe,
+             conj_cle_apply, coe_restrict_scalars', w, continuous_linear_map.zero_apply],
+end
+
+end conformal_complex_normed_space
+
+section into_the_complex_plane
+
+variables {f : ℂ → ℂ} {z : ℂ}
+
+lemma conformal_at_iff_holomorphic_or_antiholomorph_at_aux (hf : differentiable_at ℝ f z) :
+  conformal_at f z ↔
+  (differentiable_at ℂ f z ∨ differentiable_at ℂ (f ∘ conj) (conj z)) ∧ fderiv ℝ f z ≠ 0 :=
+by rw [conformal_at_iff_is_conformal_map_fderiv,
+       is_complex_or_conj_complex_linear_iff_is_conformal_map,
+       differentiable_at_iff_exists_linear_map ℝ hf,
+       antiholomorph_at_iff_exists_complex_linear_conj hf]
+
+/-- A complex function is conformal if and only if the function is holomorphic or antiholomorphic
+    with a nonvanishing differential. -/
+lemma conformal_at_iff_holomorphic_or_antiholomorphic_at :
+  conformal_at f z ↔
+  (differentiable_at ℂ f z ∨ differentiable_at ℂ (f ∘ conj) (conj z)) ∧ fderiv ℝ f z ≠ 0 :=
+iff.intro
+  (λ h, (conformal_at_iff_holomorphic_or_antiholomorph_at_aux h.differentiable_at).mp h)
+  (λ h, by
+    { have : differentiable_at ℝ f z :=
+        by_contra (λ w, h.2 $ fderiv_zero_of_not_differentiable_at w),
+      exact (conformal_at_iff_holomorphic_or_antiholomorph_at_aux this).mpr h, } )
+
```

Comment:
```suggestion
begin
  rw conformal_at_iff_is_conformal_map_fderiv,
  rw is_complex_or_conj_complex_linear_iff_is_conformal_map,
  apply and_congr_left,
  intros h,
  have h_diff := h.imp_symm fderiv_zero_of_not_differentiable_at,
  apply or_congr,
  { rw differentiable_at_iff_exists_linear_map ℝ h_diff },
  rw ← conj_conj z at h_diff,
  rw differentiable_at_iff_exists_linear_map ℝ (h_diff.comp _ conj_cle.differentiable_at),
  refine exists_congr (λ g, rfl.congr _),
  have : fderiv ℝ conj (conj z) = _ := conj_cle.fderiv,
  simp [fderiv.comp _ h_diff conj_cle.differentiable_at, this, conj_conj],
end
```

Reply:
This looks awesome! I will try to understand the proof after the CI finishes its job, and remove all those lemmas.

Reply:
I have removed the lemmas you mentioned and completed one suggested change about refines.